src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -79,6 +89,17 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         Person personToEdit = lastShownList.get(index.getZeroBased());
+
+        // Check if the person to edit is a Student and if the industry field is being edited
+        if (personToEdit instanceof Student && editPersonDescriptor.getIndustry().isPresent()) {
+            throw new CommandException(MESSAGE_CANNOT_EDIT_STUDENT_INDUSTRY);
+        }
```

Comment:
Wondrous defensive code! It is so essential to ensure that industry field belonging to company is immutable when category is student!